capstone/src/test/java/com/step902020/capstone/EventControllerTests.java

```diff
@@ -0,0 +1,198 @@
+        expectedEvent = new Event(expectedUniversity,
+                expectedOrganization.getDatastoreId(), "pizza party", "2020-08-20T12:30",
+                "Turtles bring pizza",40.769579, -73.973036, "party",
+                "2", "indoors", true, false, true);
+
+        incorrectEvent = new Event(expectedUniversity,
+                expectedOrganization.getDatastoreId(), "pizza making", "2020-08-20T12:30",
+                "Turtles bring pizza",40.769579, -73.973036, "workshop",
+                "2", "indoors", true, true, true);
+
+        // Keep old date
+        pastEvent = new Event(expectedUniversity,
+                expectedOrganization.getDatastoreId(), "pizza party", "2020-07-20T12:30",
+                "Turtles bring pizza",40.769579, -73.973036, "party",
+                "2", "indoors", true, false, true);
+
+        String individualName = individual.firstName + " " + individual.lastName;
+        reviewAddedToEvent = new Review(individualName, individual.email, "10/10 Test Review Added to Event");
+        expectedEvent.addReview(reviewAddedToEvent);
+        this.eventRepository.save(expectedEvent);
+
+        expectedReview = new Review(individualName, individual.email, "10/10 Test Expected Review");
+
+        this.authRestTemplate = this.restTemplate
+                .withBasicAuth(currentUserEmail, currentUserPassword);
+    }
+
+    @After
+    public void tearDown() {
+        this.individualRepository.deleteByEmail(expectedIndividual.getEmail());
+        this.individualRepository.deleteByEmail(currentUserEmail);
+        this.organizationRepository.deleteByEmail(expectedOrganization.getEmail());
+        this.organizationRepository.deleteByEmail(currentUserEmail);
+        this.reviewRepository.deleteByIndividualEmail(expectedIndividual.getEmail());
+        this.universityRepository.deleteByName("Test");
+        this.eventRepository.deleteByEventDateTime(expectedEvent.getEventDateTime());
+    }
+
+    @Test
+    public void testGetEvent() throws URISyntaxException {
+        // getting the actual result
+        final String baseUrl = "/get-event?event-id=" + expectedEvent.datastoreId;
+        Event result = authRestTemplate.getForObject(baseUrl, Event.class);
+        assertEquals("Wrong event returned -- id", expectedEvent.datastoreId, result.datastoreId);
+    }
+
+    @Test
+    public void testGetFilteredEvents() throws URISyntaxException {
+        // All filterable params filled (all but eventTitle)
+        Event[] result = createFilterTests(expectedUniversity.name, "", expectedEvent.eventType,
+                expectedEvent.energyLevel, expectedEvent.location, String.valueOf(expectedEvent.foodAvailable),
+                String.valueOf(expectedEvent.free), String.valueOf(expectedEvent.visitorAllowed));
+
+        assertEquals("Filtered incorrectly", 1, result.length); // Should not include pastEvent
+        assertTrue("Filtered incorrectly", result[0].equals(expectedEvent));
```

Comment:
I am not sure if this is really necessary, but these smaller tests within this large test can be separated out into separate tests.

Reply:
Yeah I was debating doing that but I didn't know if this I should group all the tests together since they all test the same function? I have no idea if that's suppose to be a good practice or not

Reply:
Yeah I was debating doing that but I didn't know if this I should group all the tests together since they all test the same function?? I have no idea if that's suppose to be a good practice or not

Reply:
+1 for smaller tests.
We often add multiple test methods for the same function like `testGetFilteredEvents_emptyFilter()`, `testGetFilteredEvents_titleFilter()`, etc.